benchmark/testsystems/bookkeepers.py

```diff
@@ -45,138 +45,114 @@ def accumulate_shadow_work(self, x_0, v_0, n_steps):
-        n_trials = ghmc.getGlobalVariableByName("ntrials")
-        return float(n_accept) / n_trials
+        xchmc = self.unbiased_simulation.integrator
+        return float(sum(xchmc.all_counts[:-1])) / sum(xchmc.all_counts)
 
-    def reset_ghmc_statistics(self):
-        """Reset the number of acceptances and number of proposals."""
-        ghmc = self.unbiased_simulation.integrator
-        ghmc.setGlobalVariableByName("naccept", 0)
-        ghmc.setGlobalVariableByName("ntrials", 0)
+    def construct_unbiased_simulation(self, use_reference=False):
+        n_steps = 10
+        return self.construct_simulation(
+            XCGHMCIntegrator(temperature=self.temperature, steps_per_hmc=n_steps, extra_chances=15,
+                             steps_per_extra_hmc=n_steps, timestep=self.xcghmc_timestep), use_reference=use_reference)
 
     def collect_equilibrium_samples(self):
         """Collect equilibrium samples, return as (n_samples, n_atoms, 3) numpy array"""
+
         print("Collecting equilibrium samples for '%s'..." % self.name)
-        # Minimize energy by gradient descent
+
+        self.unbiased_simulation = self.construct_unbiased_simulation()
+        set_positions(self.unbiased_simulation, self.positions)
         print("Minimizing...")
+        self.unbiased_simulation.minimizeEnergy()
 
-        min_sim = self.construct_simulation(GradientDescentMinimizationIntegrator())
-        min_sim.context.setPositions(self.positions)
-        min_sim.context.setVelocitiesToTemperature(self.temperature)
-        for _ in tqdm(range(100)):
-            min_sim.step(1)
-        pos = get_positions(min_sim)
-        del (min_sim)
-        gc.collect()
-
-        # "Equilibrate" / "burn-in"
-        # Running a bit of Langevin first improves GHMC acceptance rates?
-        print("Intializing with Langevin dynamics...")
-        langevin_sim = self.construct_simulation(
-            LangevinSplittingIntegrator("V R O R V", temperature=self.temperature,
-                                        collision_rate=1.0 / unit.picoseconds,
-                                        timestep=2 * self.ghmc_timestep, measure_heat=False)
-        )
-        set_positions(langevin_sim, pos)
-        for _ in tqdm(range(self.burn_in_length)):
-            langevin_sim.step(1)
-        pos = get_positions(langevin_sim)
-        del (langevin_sim)
-        gc.collect()
-
-        print('"Burning in" unbiased GHMC sampler for {:.3}ps...'.format(
-            (self.burn_in_length * self.ghmc_timestep).value_in_unit(unit.picoseconds)))
-        self.unbiased_simulation = self.construct_simulation(
-            GHMCIntegrator(temperature=self.temperature, timestep=self.ghmc_timestep))
-        set_positions(self.unbiased_simulation, pos)
+        print('"Burning in" unbiased sampler for {:.3}ps...'.format(
+            (self.burn_in_length * self.xcghmc_timestep * 10).value_in_unit(unit.picoseconds)))
```

Comment:
Approximate number of verlet steps per integrator `step`, replaced the hard-coded `10` with `self.steps_per_hmc`